### ISSUE TITLE
File decryption fails (wrong final block length)

### ISSUE THREAD

**Issue by lukeIam** [travis-ci/travis-ci#4746]
Decrypting my file fails with `bad decrypt: wrong final block length`.

I used `travis encrypt-file file` under Windows to encrypt my file without problems.
Also I added the generated decryption command to the `before_install:` section:
```
openssl aes-256-cbc -K $encrypted_048ea30036f2_key -iv $encrypted_048ea30036f2_iv -in file.enc -out file -d
```
(Also tried the `--add` option of `travis encrypt-file`)

Here are parts of the build log:
```
[...]
$ export encrypted_048ea30036f2_key=[secure]
$ export encrypted_048ea30036f2_iv=[secure]
[...]
$ openssl aes-256-cbc -K $encrypted_048ea30036f2_key -iv $encrypted_048ea30036f2_iv -in file.enc -out file -d
bad decrypt
139709116544672:error:0606506D:digital envelope routines:EVP_DecryptFinal_ex:wrong final block length:evp_enc.c:532:
The command "openssl aes-256-cbc -K $encrypted_048ea30036f2_key -iv $encrypted_048ea30036f2_iv -in file.enc -out file -d" failed and exited with 1 during .
Your build has been stopped.
```

Any suggestions?

**Comment by BanzaiMan** [comment]
Could you try steps with `travis encrypt-file -r OWNER/REPO file`?

**Comment by lukeIam** [comment]
sure - but the same problem with the same message

Some more information about the project:
we need a new build system so I'm trying out Travis CI.
-> the `.travis.yml` only exists in a brach `travisTest` in my fork [lukeIam/Vocaluxe](https://github.com/lukeIam/Vocaluxe/tree/travisTest).
Travis is enabled for this fork and was successfully building, before I added the encrypted file.
The log of the last build can be found here:
https://travis-ci.org/lukeIam/Vocaluxe/builds/78612690

**Comment by BanzaiMan** [comment]
I just remembered. You cannot encrypt more than 1 file, because the second invocation nullifies the first. This is documented. http://docs.travis-ci.com/user/encrypting-files/#Encrypting-multiple-files

**Comment by lukeIam** [comment]
@BanzaiMan Thanks for your time! But I'm afraid I don't get it...

"You cannot encrypt more than 1 file" - I knew this and I've only encrypted a single file and of course I have only one openssl line in my [.travis.yml](https://github.com/lukeIam/Vocaluxe/blob/travisTest/.travis.yml).

The other key/iv pair was from the last attempt and is now deleted.
[build log](https://travis-ci.org/lukeIam/Vocaluxe/builds/78738939)

Or does does the `secure` line of the `deploy` section (github releases), consumes the single possible openssl call, even it does't use a file?

**Comment by BanzaiMan** [comment]
@lukeIam In the build you indicated, I see 2 sets of environment variables set by `travis encrypt-file …` command (https://travis-ci.org/lukeIam/Vocaluxe/builds/78612690#L935-L938) This indicates to me that you ran the command twice, and you added those in your repository's settings.

In https://travis-ci.org/lukeIam/Vocaluxe/builds/78738939, I see that you've removed one, but perhaps you removed the wrong one.

**Comment by lukeIam** [comment]
@BanzaiMan 
> In https://travis-ci.org/lukeIam/Vocaluxe/builds/78738939, I see that you've removed one, but perhaps you removed the wrong one.

I removed all four variables with `travis env unset [...]` and reran `travis encrypt-file`.
It recreated $encrypted_a377c9765acb (same variable name).

I started an other try and got the same variable again.
Here is what I executed:
```bat
Microsoft Windows [Version 6.3.9600]
(c) 2013 Microsoft Corporation. All rights reserved.

D:\Projects\Vocaluxe>travis env list
# environment variables for lukeIam/Vocaluxe
encrypted_a377c9765acb_key=[secure]
encrypted_a377c9765acb_iv=[secure]

D:\Projects\Vocaluxe>travis env unset encrypted_a377c9765acb_key
[x] removing environment variable $encrypted_a377c9765acb_key

D:\Projects\Vocaluxe>travis env unset encrypted_a377c9765acb_iv
[x] removing environment variable $encrypted_a377c9765acb_iv

D:\Projects\Vocaluxe>travis env list
# environment variables for lukeIam/Vocaluxe

D:\Projects\Vocaluxe>travis whoami
You are lukeIam

D:\Projects\Vocaluxe>travis encrypt-file -r lukeIam/Vocaluxe travisKey
encrypting travisKey for lukeIam/Vocaluxe
storing result as travisKey.enc
storing secure env variables for decryption

Please add the following to your build script (before_install stage in your .tra
vis.yml, for instance):

    openssl aes-256-cbc -K $encrypted_a377c9765acb_key -iv $encrypted_a377c9765a
cb_iv -in travisKey.enc -out travisKey -d

Pro Tip: You can add it automatically by running with --add.

Make sure to add travisKey.enc to the git repository.
Make sure not to add travisKey to the git repository.
Commit all changes to your .travis.yml.

D:\Projects\Vocaluxe>
```

**Comment by BanzaiMan** [comment]
Hmmm. Interesting. Do you have access to another OS (Linux or Mac)? Does this happen there, too?

**Comment by lukeIam** [comment]
I'll install a linux...

**Comment by cotsog** [comment]
Maybe related: we had a [report](https://secure.helpscout.net/conversation/116660963/19564/) of a user not able to encrypt env variables on Windows.

**Comment by lukeIam** [comment]
On ubuntu 14.04 LTS x64 it worked:
https://travis-ci.org/lukeIam/Vocaluxe/builds/78758729
(The build faild, but this is caused by my deploy script - openssl is working!)

`travis encrypt-file` generated a new variable with linux

**Comment by MariadeAnton** [comment]
related to @cotsog - it was failing in a Windows 8 machine, using the default Command Prompt and ConEmu x64

**Comment by omaralvarez** [comment]
This is also happenning on Windows 7 x64 with Git Bash command prompt.

**Comment by BartLangelaan** [comment]
Also happening on Windows 10. Trying to decrypt a Google Apps Engine service account json, with no luck. Any recent updates that could have broken this?

**Comment by hstolte** [comment]
I am experiencing the same issue on Windows 10 using the default command promt.

**Comment by shizuka** [comment]
+1, using default prompt on Windows 10 Pro 1511

```
$ openssl aes-256-cbc -K $encrypted_9a78d482b28c_key -iv $encrypted_9a78d482b28c_iv -in deploy-key.enc -out deploy-key -d

bad decrypt

140405926045344:error:0606506D:digital envelope routines:EVP_DecryptFinal_ex:wrong final block length:evp_enc.c:532:

The command "openssl aes-256-cbc -K $encrypted_9a78d482b28c_key -iv $encrypted_9a78d482b28c_iv -in deploy-key.enc -out deploy-key -d" failed and exited with 1 during .

Your build has been stopped.
```

**Comment by nickvandewiele** [comment]
+1

**Comment by JiongBull** [comment]
+1 windows 10 Git Bash

**Comment by Waydrow** [comment]
+1 windows 10 Git Bash

**Comment by bterlson** [comment]
+1

**Comment by Waydrow** [comment]
Ok, finally I made it by using Linux, Ubuntu to deploy it

**Comment by waltertamboer** [comment]
@Waydrow Yup, that's what I did as well. I've installed Ubuntu on a virtual machine using VirtualBox.

But... +1

**Comment by SeayXu** [comment]
@lukeIam Do you solved the problem?

**Comment by lukeIam** [comment]
@SeayXu not really - I installed Ubuntu in a virtual box...

**Comment by SeayXu** [comment]
@lukeIam So do i,but got the same problem.
I'm using [another way](http://git.seay.me/post/2016-03-26-Hexo-With-TravisCI.html) by `github access token`

**Comment by SeayXu** [comment]
@lukeIam Last night,i try again,no problem

**Comment by TomeOkin** [comment]
I also meet the problem:
`The command "openssl aes-256-cbc -K $encrypted_e59b7992228b_key -iv $encrypted_e59b7992228b_iv -in jo.jks.enc -out jo.jks -d" failed and exited with 1 during .`

I has try cmd and git shell in win7, but get the same result.

**Comment by SuriyaaKudoIsc** [comment]
Got the same problem: https://travis-ci.org/iSCInc/status/builds/123697131!

**Comment by SeayXu** [comment]
@TomeOkin @SuriyaaKudoIsc if you don't want to use Linux system,just see[ this way](http://git.seay.me/post/2016-03-26-Hexo-With-TravisCI.html);I found the reason for this problem is that     the Travis encryption generation variable is different from the Linux system in the windows system.So,I suggest you use Linux system do this.

**Comment by SuriyaaKudoIsc** [comment]
@SeayXu Well... :sweat_smile: I can't speak or read in Chinese.

**Comment by SeayXu** [comment]
@SuriyaaKudoIsc OK,[here is English article](http://koyasu221b.com/2016/01/23/deploy-hexo-github-pages-by-travis/)

**Comment by MikeJerred** [comment]
+1 Using Windows 8.1 here and got the same error. Encrypting the file on linux (using Google Cloud Shell) works fine

**Comment by snowdream** [comment]
This is also happenning on Windows 7 x64 with Git Bash command prompt.

**Comment by trashier** [comment]
This issue still occurs.

**Comment by nawazdhandala** [comment]
+1

**Comment by Jameskmonger** [comment]
Still experiencing this on Windows.

**Comment by Bosek** [comment]
+1 Win10
